DESIGN.md

```diff
@@ -73,77 +157,225 @@ Flow control support:
 - buffering, sampling, throttling, windowing, dropping, etc
 - temporal and count-based strategies
 - `request(n)` consumer demand signal
-  - for pull-based sources, this allows batched "async pull"
-  - for push-based sources, this allows backpressure signals to conditionally apply strategies (i.e. drop, buffer, sample, fail, etc)
+	- for pull-based sources, this allows batched "async pull"
+	- for push-based sources, this allows backpressure signals to conditionally apply strategies (i.e. drop, first, buffer, sample, fail, etc)
+
+You get a flowable from:
+
+- Converting a Observable with a backpressure strategy
+- Create from sync/async OnSubscribe API (which participate in backpressure semantics)
+
+*Type Signature*
+
+```java
+class Flowable<T> implements Flow.Publisher<T>, io.reactivestreams.Publisher<T> {
+  void subscribe(Subscriber<T> subscriber);
+
+  interface Subscriber<T> implements Flow.Subscriber<T>, io.reactivestreams.Subscriber<T> {
+    void onNext(T t);
+    void onError(Throwable t);
+    void onComplete();
+    void onSubscribe(Flowable.Subscription d);
+  }
+
+  interface Subscription implements Flow.Subscription, io.reactivestreams.Subscription {
+    void cancel();
+    void request(long n);
+  }
+}
+```
+
+*NOTE: To support `Flow.Publisher` in Java 9+ without breaking Java 7+ compatibility, we want to use the [multi-release jar file support](http://openjdk.java.net/jeps/238).*
+
+The rule for using this type signature is:
+
+> onSubscribe onNext* (onError | onComplete)?
+
+
+##### Single
+
+Lazy representation of a single response (lazy equivalent of `Future`/`Promise`).
+
+Usable for:
+
+- pull sources
+- push sources being windowed or flow controlled (such as `window(1)` or `take(1)`)
+- sync or async
+- 1 item
+
+Flow control:
+
+- Not applicable (don't subscribe if the single response is not wanted)
+
+*Type Signature*
+
+```java
+class Single<T> {
+  void subscribe(Single.Subscriber<T> subscriber);
+
+  interface Subscriber<T> {
```

Comment:
Would you rather `io.reactivex.SingleSubscriber` or `io.reactivex.Single.SingleSubscriber`? I prefer the non-static inner class `SingleSubscriber`.